src/balanced_ternary_converter.cpp

```diff
@@ -28,25 +28,23 @@ std::vector<int> make_ternary(const std::vector<int>& values, int value) {
     std::vector<int> ternary(values.size(), 0);
 
     for (std::size_t i = 0; i < values.size(); ++i) {
-        for (uint8_t step = 0; step < 2; ++step) {
-            if (value >= values[i]) {
-                ++ternary[i];
-                value -= values[i];
-            }
-        }
+        ternary[i] += value / values[i];
+        value %= values[i];
     }
+    std::reverse(ternary.begin(), ternary.end());
 
```

Comment:
I redirected the way digits stored in vector.

In the usual implementation, ternary numbers are represented using vectors, with each int representing one ternary bit of the number. 

When data being read into a vector, the highest bit of the number is at the head of the vector (the position with the smallest subscript). The reason for this is that **the length of ternary numbers may change** , but we want the same weight bits to remain aligned (e.g., we want all the digits to be in subscript [0], all the tens to be in subscript [1]...); at the same time, we want all the tens to be in subscript [1]... ...); and the fact that addition, subtraction, and multiplication are generally performed from the first digit (think of vertical arithmetic in elementary school) gives a good reason for "inverted storage". And this makes benifits for further numeric computations in this project.

Reply:
It's similar to high-precision quadrature operations in competitive programming.

Reply:
I had a feeling that this would have been a better solution from the start. Not sure how I never caught this problem.